tests/unit_tests/test_runner_service.py

```diff
@@ -0,0 +1,252 @@
+from arteria.web.state import State as arteria_state
+from checksum.runner_service import Job, RunnerService
+
+import os
+import tempfile
+import pytest
+import logging
+import asyncio
+
+
+class TestJob:
+    def test_basic_command(self, caplog):
+        """
+        Test basic command is run and logged
+        """
+        job_id = 1
+        cmd = ["echo", "test"]
+        caplog.set_level(logging.INFO)
+
+        job = Job(job_id, cmd)
+        assert job.job_id == job_id
+        assert job.cmd == cmd
+
+        job.wait()
+
+        assert job.get_status() == arteria_state.DONE
+        assert caplog.records[0].levelname == "INFO"
+        assert caplog.records[0].msg == (
+                f"Starting:\n job id: {job_id}\n cmd: {cmd}")
+
+    def test_stdout_to_file(self):
+        """
+        Test it is possible to redirect stdout to a file.
+        """
+        job_id = 2
+        msg = "test"
+        cmd = ["echo", msg]
+        stdout = tempfile.NamedTemporaryFile(mode='r')
+        job = Job(job_id, cmd, stdout=stdout)
+
+        job.wait()
+        stdout.seek(0)
+
+        assert job.get_status() == arteria_state.DONE
+        assert stdout.read() == f"{msg}\n"
+
+    def test_set_cwd(self):
+        """
+        Test it is possible to set the running directory.
+        """
+        job_id = 3
+        filename = "test.txt"
+        temp_dir = tempfile.TemporaryDirectory()
+        cmd = ["touch", filename]
+        job = Job(job_id, cmd, cwd=temp_dir.name)
+
+        job.wait()
+
+        assert os.path.exists('/'.join([temp_dir.name, filename]))
+
+    def test_log_and_raise_exc(self, caplog):
+        """
+        Test an exception is raised when an error occurs in the command, and
+        that the error is logged.
+        """
+        job_id = 4
+        cmd = ["fakecmd"]
+        caplog.set_level(logging.INFO)
+
+        with pytest.raises(Exception):
+            Job(job_id, cmd)
+
+        assert caplog.records[-1].levelname == "ERROR"
+
+    def test_cancel(self, caplog):
+        """
+        Test it is possible to cancel a job and that this action is logged
+        """
+        job_id = 5
+        cmd = ["sleep", "60"]
+        caplog.set_level(logging.INFO)
+
+        job = Job(job_id, cmd)
+
+        assert job.get_status() == arteria_state.STARTED
+
+        job.cancel()
+
+        assert job.get_status() == arteria_state.CANCELLED
+        assert caplog.records[-1].levelname == "INFO"
+        assert caplog.records[-1].msg == f"Cancelling job {job_id} (`{cmd}`)"
+
+
+class TestRunnerService:
+    def test_constructor(self):
+        """
+        Test is it possible to build a service with the given history length.
+        """
+        history_len = 5
+        checksum_service = RunnerService(history_len)
+
+        assert checksum_service._job_history.maxlen == history_len
+
+    @pytest.mark.asyncio
+    async def test_generate_next_id(self):
+        """
+        Test no two generated ids are the same.
```

Comment:
Yes exactly, it's fine to have twice the same id if the service was restarted in between. What we want to avoid is having the same id twice in the current history list.